src/wp-includes/class-wp-scripts.php

```diff
@@ -302,11 +302,19 @@ public function do_item( $handle, $group = false ) {
 			$ver = $ver ? $ver . '&amp;' . $this->args[ $handle ] : $this->args[ $handle ];
 		}
 
-		$src         = $obj->src;
-		$strategy    = $this->get_eligible_loading_strategy( $handle );
-		$cond_before = '';
-		$cond_after  = '';
-		$conditional = isset( $obj->extra['conditional'] ) ? $obj->extra['conditional'] : '';
+		$src               = $obj->src;
+		$strategy          = $this->get_eligible_loading_strategy( $handle );
+		$intended_strategy = $this->get_data( $handle, 'strategy' );
+		$cond_before       = '';
+		$cond_after        = '';
+		$conditional       = isset( $obj->extra['conditional'] ) ? $obj->extra['conditional'] : '';
+
+		if (
+			empty( $intended_strategy )
+			|| ( 'async' !== $intended_strategy && 'defer' !== $intended_strategy )
+		) {
```

Comment:
```suggestion
		if ( ! $this->is_delayed_strategy( (string) $intended_strategy ) ) {
```

Reply:
`WP_Scripts::add_data()` already validates that a strategy is valid before adding the 'strategy' data, so we probably only need to check for if `! $intended_strategy`. No complaints about the extra check though if we want this to be more robust against edge cases.

Reply:
Someone _could_ try manually injecting bad strategies like so:

```php
wp_scripts()->registered[ $handle ]->extra['strategy'] = 'evil';
```

I did this in a test actually:

https://github.com/10up/wordpress-develop/blob/575c8373ce45f911bc00c9ccad6afa61656e5ed6/tests/phpunit/tests/dependencies/scripts.php#L533

Reply:
I've committed your suggestions thanks @westonruter